src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue

```diff
@@ -1,300 +1,57 @@
-      } catch (e) {
-        if (e instanceof SyntaxError || e.name === 'SyntaxError') {
-          return e
-        }
-      }
-
-      const hasMembers = some(ast, isObject)
-      return hasMembers
-        ? ast
-        : null
-    },
-
-    doesContentContainExpression() {
-      return !!this.contentExpressionAST
-    },
-
-    doesContentContainExpressionError() {
-      return !!(this.contentExpressionAST instanceof Error)
-    },
-
-    characterCounter() {
-      const
-        hasUnicode = !/^[\x00-\x7F]*$/.test(this.content)
-      const count = this.content.length
-
-      console.debug('BlockTextContentEditorForLangAndType', 'characterCounter', {hasUnicode, count})
-
-      return {
-        hasUnicode,
-        count,
-        pages: hasUnicode ? Math.ceil(count / 70) : Math.ceil(count / 160),
-      }
-    },
-  },
-
-  methods: {
-    ...mapActions('flow', ['resource_setOrCreateValueModeSpecific']),
-
-    select() {
-      this.isSelected = true
-    },
-    deselect() {
-      this.isSelected = false
-    },
-
-    // debouncedSaveTree: debounce(function () {
-    //   this.$store.dispatch('attemptSaveTree')
-    // }, 500),
-  },
-}
-</script>
-
-<style lang="scss" scoped>
-.resource-variant-text-editor {
-  margin-bottom: 10px;
-}
-
-.block-text-content-editor-for-lang-and-type {
-  margin-bottom: 0.5em;
-
-  .content-editor {
-    position: relative;
-
-    textarea {
-      height: 56px;
-    }
-
-    textarea, input[type="text"] {
-      &:not(:focus) {
-        + button {
-          opacity: 1;
-        }
-      }
-
-      + button {
-        position: absolute;
-        bottom: 7px;
-        // 15px padding => parent, 8px padding => textarea
-        right: 10px; //15px + 7px;
-
-        transition: opacity 200ms ease-in-out;
-        opacity: 0;
-      }
-    }
-
-    &-selected {
-      textarea, input[type="text"] {
-        background-color: #F8F8F8;
-        //border-color: #DDDDDD;
-      }
-    }
+<script lang="ts">
+import {Component, Prop} from 'vue-property-decorator'
+import {mixins} from 'vue-class-component'
+import Lang from '@/lib/filters/lang'
+import {IResource, IResourceValue} from '@floip/flow-runner'
+import {namespace} from 'vuex-class'
+import ExpressionInput from '@/components/common/ExpressionInput.vue'
+
+const flowVuexNamespace = namespace('flow')
+
+@Component({
+  components: {ExpressionInput},
+})
+class AdvancedExitEditor extends mixins(Lang) {
+  @Prop() readonly resourceId!: IResource['uuid']
+  @Prop({default: ''}) readonly label!: string
+  @Prop() readonly placeholder!: string
+  @Prop() readonly resourceVariant!: IResourceValue
+  @Prop() readonly mode!: string
+  @Prop({default: true}) readonly enableAutogenButton!: boolean
+  @Prop({default: 2}) readonly rows!: number
+
+  get content(): string {
+    return this.resourceVariant.value
   }
 
-  .content-toolbar {
-    margin-top: 1px;
+  commitExpressionChange(value: string): void {
+    const {resourceId, mode} = this
```

Comment:
If we used destruction approach for resourceId, mode ... I'd suggest we put `resourceVariant` here as well. Then change `this.resourceVariant` with `resourceVariant` for all below codes

Reply:
Done!